### PR TITLE
Add endpoint to record user page views

### LINKED ISSUE #12: Trackdown page view activity
We would like a table to record user pageview activities so that we can retrieve most popular articles in the future

- [x] Update db schema for the new table PageView

    - Fields: userId, storyId, createdAt
    - After creating schema, run `npx prisma generate`

- [x] Create POST endpoint for the new table

- [x] Update API docs

### PR DESCRIPTION
**Pull request by yeshengliu**
# Description

Add schema, endpoint to record user page view activity

~~The endpoint requires future update which needs information on user_id (logged in user, guest, etc), right now I put a placeholder in that field.~~

Modified `user_id` as required body param, require to call `useSession()` at front-end to retrieve `userId` before making API call

Closes #12 

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

url: `http://localhost:3000/api/user/page_views`

| body | Output | Scenario |
| --- | --- | --- |
| `none` | 400 bad request | missing req body |
| `{ "story_id": "abc" }` | 400 bad request | invalid story_id |
| `{ "storyId": "6488c516f5f617c772f6f60e" }` | 400 bad request | missing story_id field |
| `{ "story_id": "6488c516f5f617c772f6f60e" }` | 400 bad request | story_id is valid, missing user_id field |
| `{ "story_id": "6488c516f5f617c772f6f601", "user_id": "647ad6fda9efff3abe83044f" }` | 404 not found | nonexistent story_id |
| `{ "story_id": "6488c516f5f617c772f6f60e", "user_id": "647ad6fda9efff3abe83044f" }` | 200 ok | success request |

### CODE REVIEW

File: src/app/api/user/page_views/route.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (story === null) {
+      return NextResponse.json({ error: "Story not found" }, { status: 404 });
+    }
+  } catch (e) {
+    return NextResponse.json({ error: e }, { status: 400 });
+  }
+
+  //TODO: retrieve userId from session/cookie
+  const userId = "647ad6fda9efff3abe83044f";
```
Comment (hdavidethan) [inline src/app/api/user/page_views/route.ts:28]: TODO: retrieve userId from session/cookie

File: src/app/api/user/page_views/route.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import prisma from "@/lib/prisma";
+import { NextResponse } from "next/server";
+
+export async function POST(req: Request) {
+  const { searchParams } = new URL(req.url);
+  const storyId = searchParams.get("story_id");
```
Comment (hdavidethan) [inline src/app/api/user/page_views/route.ts:6]: For POST requests, we prefer using body parameters rather than path parameters.

**Review by hdavidethan — CHANGES_REQUESTED** [review]

File: src/app/api/user/page_views/route.ts (hunk 2 above)
Comment (yeshengliu) [reply src/app/api/user/page_views/route.ts]: Fixed the issue

File: src/app/api/user/page_views/route.ts (hunk 1 above)
Comment (yeshengliu) [reply src/app/api/user/page_views/route.ts]: Accept `userId` as body param. Close #16

**Review by hdavidethan — APPROVED** [review]

**Pull request merged**